### ISSUE TITLE
Offer to 'Find all references' on overridden method

### ISSUE THREAD

**Issue by raboof** [scalameta/metals#1063]
When doing a 'find all references' (in vscode) on a method that overrides a method from a superclass, only references to the method on the subclass are returned. It would be useful to have the option to also see references to the method on the superclass. Also easily jumping to the method in the superclass would be helpful.

**Comment by olafurpg** [comment]
Moving to the main repo since this issue is unblocked by textDocument/rename.

**Comment by raboof** [comment]
> I think it's fine to default to always finding references to all overridden and supermethods.

yeah, only for `toString` and similar generic methods it can be helpful to only search for the specific one, for now I agree it makes sense to search for the supermethods

**Comment by olafurpg** [comment]
Good point. We may want to special case `toString`, `hashCode` and `equals` to not include supermethods and overridden methods.

**Comment by kubukoz** [comment]
Hi, is anyone currently working on this? If it's not too hard for a newcomer to the project, I'd like to give it a try :)

**Comment by tgodzik** [comment]
> Hi, is anyone currently working on this? If it's not too hard for a newcomer to the project, I'd like to give it a try :)

Not currently, you're welcome to give it a try! We already try to find all references in the RenameProvider, which is done by going to the top symbol and searching for references of all implementations. This might probably be done better and we can move that logic to the ReferenceProvider.

**Comment by tgodzik** [comment]
@kubukoz Are you working on this? Or is ok for someone else to pick it up?

**Comment by kubukoz** [comment]
I'm not, got swamped with other commitments, sorry.

**Comment by tgodzik** [comment]
We'll work on it with @kbochenek to get him introduced into Metals and overall Scalameta environment.

**Comment by tgodzik** [comment]
This is blocked until we have a faster way to extract info about class hierarchy in dependencies.

Looking into ways to do it here: https://github.com/scalameta/metals/issues/1287
